### PR TITLE
Add more correction of OpenZeppelin/creed found issues

### PR DESCRIPTION
**Pull request by ilitteri**

### CODE REVIEW

**Review by juanbono — APPROVED** [review]

File: claim_contracts/src/ClaimableAirdrop.sol [hunk 1]
```diff
@@ -42,34 +55,35 @@ contract ClaimableAirdrop is
...
-    ) public initializer {
+    ) external initializer {
         require(_owner != address(0), "Invalid owner address");
         require(
             _tokenProxy != address(0) && _tokenProxy != address(this),
             "Invalid token contract address"
         );
         require(
-            _claimSupplier != address(0) && _claimSupplier != address(this),
+            _tokenDistributor != address(0) &&
```
Comment (MegaRedHand) [inline claim_contracts/src/ClaimableAirdrop.sol:74]: Should we remove these checks? At least the `address(0)` one is included in `__Ownable_init`

**Review by MegaRedHand — APPROVED** [review]

**Review by jrchatruc — APPROVED** [review]

File: claim_contracts/src/ClaimableAirdrop.sol (hunk 1 above)
Comment (ilitteri) [reply claim_contracts/src/ClaimableAirdrop.sol:74]: It is not strictly necessary

**Comment by ilitteri** [comment]
Fixes OpenZeppelin issues:
- L-02
- L-03
- N-03
- N-04
- N-05
- N-07
- N-08
- N-10
- N-11
- N-13

**Comment by ilitteri** [comment]
Fixes Creed issues:
- Incorrect Usage of ReentrancyGuard instead of ReentrancyGuardUpgradeable
- Redundant Modifiers in pause and unpause Functions
- Missing Events in State-Changing Functions
- Simplify Initialization of ClaimableAirdrop

**Pull request merged**